distributed/worker.py

```diff
@@ -2678,6 +2678,7 @@ def check_pause(memory):
                         if self.memory_limit is not None
                         else "None",
                     )
+                    await self.close_gracefully(restart=True)
```

Comment:
I'm trying to anticipate cases where restarting might not be welcome.  I can think of a couple:

1.  There is some slow cleanup process that needs a couple of seconds to clear memory out (I haven't seen this personally).   Maybe it makes sense to wait a few times before restarting?  This might not be worth the complexity though
2.  The worker is currently working on something, and we should give it a moment to try to finish it.  This could be determined by checking the `self.executing` set of currently executing tasks.  If this is empty then sure, restarting seems ok.  If not, then we might want to wait for a bit, but not forever.